MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -2239,29 +2239,38 @@ private PartInUse getPartInUse(Part p) {
         return (null != result.getPartToBuy()) ? result : null;
     }
 
-    private void updatePartInUseData(PartInUse piu, Part p) {
-        if ((p.getUnit() != null) || (p instanceof MissingPart)) {
+    private void updatePartInUseData(PartInUse piu, Part p, boolean ignoreMothballedUnits,
+            PartQuality ignoreSparesUnderQuality) {
+
+        if (ignoreMothballedUnits && (null != p.getUnit()) && p.getUnit().isMothballed()) {
+            // Ignore it
+        } else if ((p.getUnit() != null) || (p instanceof MissingPart)) {
             piu.setUseCount(piu.getUseCount() + getQuantity(p));
         } else {
             if (p.isPresent()) {
-                piu.setStoreCount(piu.getStoreCount() + getQuantity(p));
-                piu.addSpare(p);
+                if (p.getQuality().toNumeric() < ignoreSparesUnderQuality.toNumeric()) {
+                    // Ignore it
+                } else {
+                    piu.setStoreCount(piu.getStoreCount() + getQuantity(p));
+                    piu.addSpare(p);
+                }
             } else {
                 piu.setTransferCount(piu.getTransferCount() + getQuantity(p));
             }
         }
     }
 
     /** Update the piu with the current campaign data */
```

Comment:
This JavaDoc needs to be updated to include the variable names. Ideally it would be a little more descriptive, too.